willa/tind/fetch.py

```diff
@@ -62,3 +68,86 @@ def fetch_file(file_url: str, output_dir: str = '') -> str:
         raise RecordNotFoundError('Referenced file could not be downloaded.')
 
     return saved_to
+
+def fetch_file_metadata(record: str) -> Record:
+    """Fetch file metadata for a given Tind record.
+    :raises AuthorizationError: When the TIND API key is invalid.
+    :raises Exception: for any response other then 200. 
+    :returns a json list of metadata for a given Tind record
+    """ 
+
+    status, files = tind_get(f"record/{record}/files")
+
+    if status != 200 and status < 500:
+        j = json.loads(files)
+        reason = j['reason']
+        raise TINDError(f"Status: {status} Message: {reason}.")
+    
+    return json.loads(files)
+
+def fetch_ids_search(search: str):
+    status, rec_ids = tind_get(f"search", {'p': search})
+
+    if status >= 400:
+        j = json.loads(rec_ids)
+        reason = j['reason']
+        raise HTTPError(f"Status: {status}, Message: {reason}")
+
+    j = json.loads(rec_ids) 
+    return j['hits']
+
+def fetch_marc_by_ids(ids: list):
+    """Fetch Tind marc from a list of Tind record ids
+    :returns a list of PYMARC records 
+    """ 
+    records = []
+    for item in ids:
+      m = fetch_metadata(item)
+      records.append(m)
+
+    return records
+
+def fetch_search_metadata(search: str):
+    ids = fetch_ids_search(search)
+
+    return fetch_marc_by_ids(ids)
+
+def search_request(search: str, search_id=None ):
```

Comment:
Remove space, needs a docstring.

```suggestion

def search_request(search: str, search_id=None):
```